### PR TITLE
port some of citadel's lighting changes

### PR DESCRIPTION
**Pull request by Effectism**
Looks quite a bit prettier, and apparently improves performance somewhat.
Lighting is a lot darker now, especially at night, but it doesn't seem to be a huge issue on other servers that use this.

## Changelog
:cl: deathride58
tweak: Lighting now uses a linear algorithm to calculate falloff instead of an inverse-square algorithm.
/:cl:

### CODE REVIEW

File: code/modules/lighting/lighting_source.dm [hunk 1]
```diff
@@ -1,6 +1,12 @@
 // This is where the fun begins.
 // These are the main datums that emit light.
 
+// Thanks to Lohikar for flinging this tiny bit of code at me, increasing my brain cell count from 1 to 2 in the process.
+// This macro will only offset up to 1 tile, but anything with a greater offset is an outlier and probably should handle its own lighting offsets.
+// Anything pixelshifted 16px or more will be considered on the next tile.
+#define GET_APPROXIMATE_PIXEL_DIR(PX, PY) ((!(PX) ? 0 : ((PX >= 16 ? EAST : (PX <= -16 ? WEST : 0)))) | (!PY ? 0 : (PY >= 16 ? NORTH : (PY <= -16 ? SOUTH : 0))))
+#define UPDATE_APPROXIMATE_PIXEL_TURF var/_mask = GET_APPROXIMATE_PIXEL_DIR(top_atom.pixel_x, top_atom.pixel_y); pixel_turf = _mask ? (get_step(source_turf, _mask) || source_turf) : source_turf
```
Comment (MCterra10) [inline code/modules/lighting/lighting_source.dm]: `/* austation start`
and
`austation end`
(on new lines by the way)

File: code/modules/lighting/lighting_source.dm [hunk 2]
```diff
@@ -110,7 +116,26 @@
...
+// but results in lights having a slightly larger range, which is most noticable with large light sources.
+// This also results in lights being diamond-shaped, fuck. This looks the darkest out of the
+// three due to how lights are brighter closer to the source compared to the original falloff
+// algorithm. This falloff method also does not at all take into account lighting height, as
+// it acts as a flat reduction to light range with this method.
+//#define LUM_FALLOFF(C, T) (1 - CLAMP01(((abs(C.x - T.x) + abs(C.y - T.y))) / max(1, light_range+1)))
+
+// Linear lighting falloff but with an octagonal shape in place of a diamond shape (from CIT)
+#define LUM_FALLOFF(C, T) (1 - CLAMP01(max(GET_LUM_DIST(abs(C.x - T.x), abs(C.y - T.y)),LIGHTING_HEIGHT) / max(1, light_range+1)))
+#define GET_LUM_DIST(DISTX, DISTY) (DISTX + DISTY + abs(DISTX - DISTY)*0.4)
```
Comment (MCterra10) [inline code/modules/lighting/lighting_source.dm]: `/* austation start`
and
`austation end`
(on new lines by the way)

File: code/modules/lighting/lighting_source.dm [hunk 3]
```diff
@@ -296,3 +321,5 @@
 #undef LUM_FALLOFF
 #undef REMOVE_CORNER
 #undef APPLY_CORNER
+#undef GET_APPROXIMATE_PIXEL_DIR
+#undef UPDATE_APPROXIMATE_PIXEL_TURF
```
Comment (MCterra10) [inline code/modules/lighting/lighting_source.dm]: `/* austation start`
and
`austation end`
(on new lines by the way)

**Comment by Effectism** [comment]
webedit gang

**Comment by MCterra10** [comment]
reee follow the format for /* austation begin specified in the readme for austation

**Comment by MCterra10** [comment]
If you could add some screenshots that would be great. Going to request player review

**Comment by MCterra10** [comment]
ebic, let's hope nobody changes lighting in the near future

**Comment by MCterra10** [comment]
whoopsie forgot to squash merge

**Pull request merged**